compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2034,6 +2037,8 @@ private SyntaxKind getExpectedTokenKind(ParserRuleContext ctx) {
                 return SyntaxKind.FINAL_KEYWORD;
             case CONST_DECL_TYPE:
                 return SyntaxKind.IDENTIFIER_TOKEN;
+            case TYPEOF_KEYWORD:
+                return SyntaxKind.TYPEOF_KEYWORD;
```

Comment:
Need to add a case for `getNextRule` as well.

Try this one: `int a = 3  typeof  + 2;`.  Might fail.

Reply:
Done

Reply:
shall we add a test-case too for this particular scenario?

Reply:
test case added for `int a = 3 typeof 2;`